client/pwa/src/db.ts

```diff
@@ -57,9 +91,51 @@ export class MDNOfflineDB extends Dexie {
       watched: "url, title, path",
       notifications: "id, title, text, url, created, read, starred",
     });
+    this.version(2).stores({
+      contentStatusHistory: "++id",
+    });
   }
 }
 
 const offlineDb = new MDNOfflineDB();
 
-export { offlineDb };
+async function getContentStatus(): Promise<ContentStatus> {
+  const current = await offlineDb.contentStatusHistory.toCollection().last();
+
+  return (
+    current || {
+      phase: ContentStatusPhase.INITIAL,
+      local: null,
+      remote: null,
+      progress: null,
+      timestamp: new Date(),
+    }
+  );
+}
+
+async function patchContentStatus(
+  changes: Omit<Partial<ContentStatus>, "id" | "timestamp">
+) {
+  const oldStatus = await getContentStatus();
+  const newStatus = {
+    ...oldStatus,
+    ...changes,
+    id: undefined,
+    timestamp: new Date(),
+  };
+
+  if (newStatus.phase === ContentStatusPhase.INITIAL) {
```

Comment:
A side-effect of persisting the status more reliably in the IndexedDB is that we get stuck when the Service Worker gets stuck during non-idle phase. So we need a strategy to recover from that.

Reply:
I guess the way to do this it to have the state in both memory of the service worker and IndexedDB and if their out of sync resetting it to initial (as this means the sw got killed).

Reply:
Fixed in c9f69759d.